controllers/lb/controller.go

```diff
@@ -130,36 +118,42 @@ func (r *ServiceReconciler) SetupWithManager(mgr ctrl.Manager) error {
 }
 
 func (r *ServiceReconciler) Reconcile(req ctrl.Request) (ctrl.Result, error) {
-	_ = context.Background()
-	_ = r.Log.WithValues("porter", req.NamespacedName)
-	// your logic here
-	// Fetch the Service instance
-	err := retry.RetryOnConflict(retry.DefaultRetry, func() error {
-		return r.reconcile(req.NamespacedName)
-	})
+	log := r.Log.WithValues("porter", req.NamespacedName)
+
+	svc := &corev1.Service{}
+	err := r.Get(context.TODO(), req.NamespacedName, svc)
 	if err != nil {
-		switch t := err.(type) {
-		case portererror.ResourceNotEnoughError:
-			r.Log.Info(t.Error() + ", waiting for requeue")
-			return ctrl.Result{
-				RequeueAfter: time.Second * 10,
-			}, nil
-		case portererror.EIPNotFoundError:
-			r.Log.Info("Detect unknown ips in annotations")
-			return ctrl.Result{}, r.clearAnnotation(req.NamespacedName)
-		case portererror.EIPProtocolNotFoundError:
-			r.Log.Info(err.Error())
+		if errors.IsNotFound(err) {
+			return ctrl.Result{}, nil
+		}
+		return ctrl.Result{}, err
+	}
+
+	deleted, err := r.useFinalizerIfNeeded(svc)
+	if err == nil {
+		if deleted {
+			return ctrl.Result{}, nil
+		}
+		err = r.createLB(svc)
+	}
+
+	if err != nil {
+		switch portererror.ReasonForError(err) {
+		case portererror.EIPNotExist:
+			r.Event(svc, corev1.EventTypeWarning, "LB Created", "Cann't assign EIP, please check eip and protocol")
+			return ctrl.Result{}, r.updateService(svc, r.findEIP(svc), true)
+		case portererror.ParaInvalidError:
 			return ctrl.Result{}, nil
 		default:
 			if errors.IsNotFound(err) {
-				r.Log.Info("Maybe sevice has been deleted, skipping reconciling")
+				log.Info("Maybe sevice has been deleted, skipping reconciling")
 				return ctrl.Result{}, nil
 			}
-			r.Log.Error(t, "Create LB for service failed")
-			return ctrl.Result{RequeueAfter: time.Second * 10}, t
+			return ctrl.Result{}, err
 		}
 	}
-	return ctrl.Result{}, nil
+
+	return ctrl.Result{RequeueAfter: time.Second * 60}, nil
```

Comment:
How about set `RequeueAfter` with command line parameter

Reply:
Yes,  many params are hard code before, and should be add to commandline, this will do later.